### PR TITLE
Avoid force rediscovery during FL - Event/WFM synchronization process.

### LINKED ISSUE #784: Event/wfm topology false positive ISL detection after FL "reconnect"
Event/wfm topology track availability of FL (to not ISL fail for all known ISL when FL goes down). When "connection" to FL recovered (after it have being lost) event/wfm start cache sync process.

During this process it populate `DiscoveryManager` with existing switch and ports. It call methods `handleSwitchUp` and `handlePortUp`, one of the actions inside this methods is `subject.setFoundIsl(false)`... And it lead to false positive ISL discovery after the sync process when it over.

### PR DESCRIPTION
**Pull request by sergii-iakovenko**
The changes:
- Introduce dedicated payload entities for messages used in FL - Event/WFM synchronization: NetworkDumpSwitchData, NetworkDumpPortData.
- Use NetworkDumpSwitchData, NetworkDumpPortData as the result of execution of NetworkCommandData in FL.
- Modify OFELinkBolt to do not deactivate ISLs on NetworkDumpPortData.

Fixes #784

### CODE REVIEW

File: services/src/messaging/src/main/java/org/openkilda/messaging/info/discovery/NetworkDumpPortData.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import com.fasterxml.jackson.databind.PropertyNamingStrategy.SnakeCaseStrategy;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import lombok.Value;
+
+/**
+ * Defines the {@link InfoMessage} payload representing a switch port for network sync process.
+ */
+@Value
+@JsonNaming(SnakeCaseStrategy.class)
+public class NetworkDumpPortData extends InfoData {
```
[the remark this answers is not in the document]
Comment (sergii-iakovenko) [reply services/src/messaging/src/main/java/org/openkilda/messaging/info/discovery/NetworkDumpPortData.java]: I would agree if the suffix wasn't used for payload classes of InfoMessage (a half of classes that extend InfoData has the suffix "Data", while the another half uses "Response" instead).

So we need to either follow the existing practice or agree on a new one.

File: services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/converter/IOFSwitchConverter.java [hunk 2]
```diff
@@ -18,15 +35,26 @@
 public final class IOFSwitchConverter {
 
     /**
-     * Transforms {@link IOFSwitch} to object that is used throughout kilda in all components.
+     * Transforms {@link IOFSwitch} to {@link NetworkDumpSwitchData} object.
+     *
      * @param sw switch data.
+     * @return converted switch.
+     */
+    public static NetworkDumpSwitchData convertToNetworkDumpSwitchData(IOFSwitch sw) {
```
[the remark this answers is not in the document]
Comment (sergii-iakovenko) [reply services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/converter/IOFSwitchConverter.java]: I'll move the builder method to RecordHandler class where it's used.

File: services/wfm/src/main/java/org/openkilda/wfm/topology/event/OFELinkBolt.java [hunk 3]
```diff
@@ -366,11 +370,16 @@ private void dispatchWaitSync(Tuple tuple, InfoMessage infoMessage) {
 
     private void dispatchSyncInProgress(Tuple tuple, InfoMessage infoMessage) {
         InfoData data = infoMessage.getData();
-        if (data instanceof SwitchInfoData) {
-            handleSwitchEvent(tuple, (SwitchInfoData) data);
-        } else if (data instanceof PortInfoData) {
-            handlePortEvent(tuple, (PortInfoData) data);
-        } else if (data instanceof NetworkSyncEndMarker) {
+        if (data instanceof NetworkDumpSwitchData) {
+            logger.info("DISCO Sync: switch {}", data);
```
[the remark this answers is not in the document]
Comment (sergii-iakovenko) [reply services/wfm/src/main/java/org/openkilda/wfm/topology/event/OFELinkBolt.java]: Good point. Will change it.

File: services/src/messaging/src/main/java/org/openkilda/messaging/info/event/SwitchInfoData.java [hunk 4]
```diff
@@ -17,18 +17,22 @@
...
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.annotation.JsonPropertyOrder;
 import com.fasterxml.jackson.databind.annotation.JsonSerialize;
-import org.openkilda.messaging.info.CacheTimeTag;
 
 import java.util.Map;
 import java.util.Objects;
 
 /**
  * Defines the payload payload of a Message representing a switch info.
```
[the remark this answers is not in the document]
Comment (sergii-iakovenko) [reply services/src/messaging/src/main/java/org/openkilda/messaging/info/event/SwitchInfoData.java]: Will fix.

File: services/src/messaging/src/main/java/org/openkilda/messaging/info/discovery/NetworkDumpSwitchData.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+import lombok.Builder;
+import lombok.Value;
+
+/**
+ * Defines the {@link InfoMessage} payload representing a switch for network sync process.
+ */
+@Value
+@Builder
+@JsonNaming(SnakeCaseStrategy.class)
+public class NetworkDumpSwitchData extends InfoData {
```
[the remark this answers is not in the document]
Comment (sergii-iakovenko) [reply services/src/messaging/src/main/java/org/openkilda/messaging/info/discovery/NetworkDumpSwitchData.java]: Same as above.

File: services/src/messaging/src/main/java/org/openkilda/messaging/info/discovery/NetworkDumpPortData.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    private static final long serialVersionUID = 1L;
+
+    private String switchId;
```
[the remark this answers is not in the document]
Comment (sergii-iakovenko) [reply services/src/messaging/src/main/java/org/openkilda/messaging/info/discovery/NetworkDumpPortData.java]: Not required here as JsonNaming is used on the class.

I see that getting rid of manual declaration of @JsonProperty / @JsonCreator and using integration of Lombok + Jackson is the solution to make json serialization/deserialization "solid" and error proof. #880

**Comment by sergii-iakovenko** [comment]
The review findings have been addressed.

File: services/src/messaging/src/main/java/org/openkilda/messaging/info/discovery/NetworkDumpPortData.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (surabujin) [reply services/src/messaging/src/main/java/org/openkilda/messaging/info/discovery/NetworkDumpPortData.java]: You use `@JsonProperty` in constructor arguments and don't use it on properties itself. It make whole declaration not explicit and not obvious.

All current declarations use `@JsonProperty` on properties itself and on constructor arguments. We must follow existing practice or agree on a new one.

**Comment by surabujin** [comment]
C/FLOW and C/STATS are affected because they read "speaker" topic and due to change in messaging.

File: services/src/messaging/src/main/java/org/openkilda/messaging/info/discovery/NetworkDumpPortData.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (sergii-iakovenko) [reply services/src/messaging/src/main/java/org/openkilda/messaging/info/discovery/NetworkDumpPortData.java]: @JsonProperty is required in constructors and builder methods, unless we implement #880 

@JsonProperty on fields is optional and in the most cases being used if specific name needs to be defined. No annotation means default Jackson behavior and it's obvious and understandable. Just need to know how Jackson works.

But I agree that we should share this with the team, so here's a ticket - #1012

**Pull request merged**